File: sdk/ruby/test/wire/inboxes_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class InboxesWireTest < WireMockTestCase
  def setup
    super

    @client = AgentMailer::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_inboxes_list_with_wiremock
    test_id = "inboxes.list.0"

    @client.inboxes.list(request_options: {
      additional_headers: {
        "X-Test-Id" => "inboxes.list.0"
      }
    })

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/inboxes",
      query_params: nil,
      expected: 1
    )
  end

  def test_inboxes_create_with_wiremock
    test_id = "inboxes.create.0"

    @client.inboxes.create(request_options: {
      additional_headers: {
        "X-Test-Id" => "inboxes.create.0"
      }
    })

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/inboxes",
      query_params: nil,
      expected: 1
    )
  end

  def test_inboxes_get_with_wiremock
    test_id = "inboxes.get.0"

    @client.inboxes.get(
      inbox_id: "inboxId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "inboxes.get.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/inboxes/inboxId",
      query_params: nil,
      expected: 1
    )
  end

  def test_inboxes_delete_with_wiremock
    test_id = "inboxes.delete.0"

    @client.inboxes.delete(
      inbox_id: "inboxId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "inboxes.delete.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "DELETE",
      url_path: "/v1/inboxes/inboxId",
      query_params: nil,
      expected: 1
    )
  end

  def test_inboxes_update_with_wiremock
    test_id = "inboxes.update.0"

    @client.inboxes.update(
      inbox_id: "inboxId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "inboxes.update.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "PATCH",
      url_path: "/v1/inboxes/inboxId",
      query_params: nil,
      expected: 1
    )
  end

  def test_inboxes_issue_credentials_with_wiremock
    test_id = "inboxes.issue_credentials.0"

    @client.inboxes.issue_credentials(
      inbox_id: "inboxId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "inboxes.issue_credentials.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/inboxes/inboxId/credentials",
      query_params: nil,
      expected: 1
    )
  end
end

File: sdk/rust/tests/inboxes_test.rs
use agentmailer::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_inboxes_list_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .inboxes
        .list(
            &InboxesListQueryRequest {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/inboxes", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_inboxes_create_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .inboxes
        .create(
            &InboxCreate {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/inboxes", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_inboxes_get_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client.inboxes.get(&"inboxId".to_string(), None).await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/inboxes/inboxId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_inboxes_delete_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client.inboxes.delete(&"inboxId".to_string(), None).await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("DELETE", "/v1/inboxes/inboxId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_inboxes_update_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .inboxes
        .update(
            &"inboxId".to_string(),
            &InboxUpdate {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("PATCH", "/v1/inboxes/inboxId", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_inboxes_issue_credentials_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .inboxes
        .issue_credentials(&"inboxId".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/inboxes/inboxId/credentials", None, 1)
        .await
        .unwrap();
}
